src/subprocess.rs

```diff
@@ -1,32 +1,44 @@
-use std::os::unix::process::ExitStatusExt as _;
-
 /// An error representing a subprocess that errored
 ///
 /// This can be used to propogate a subprocesses exit status.
 /// When this error is encountered the cli will exit with the status code
 /// instead of printing an error,
 #[derive(Debug)]
-pub struct ExitStatusError {
-    status: Option<i32>,
+pub enum ExitStatusError {
+    ExitCode(i32),
+    Signal(i32),
+    Unknown,
 }
 
 impl ExitStatusError {
-    #[cfg(windows)]
+    #[cfg(unix)]
     pub(crate) fn new(status: std::process::ExitStatus) -> Self {
-        Self {
-            status: status.code(),
+        use std::os::unix::process::ExitStatusExt as _;
+
+        if let Some(code) = status.code() {
+            Self::ExitCode(code)
+        } else if let Some(signal) = status.signal() {
+            Self::Signal(signal)
+        } else {
+            Self::Unknown
         }
     }
 
-    #[cfg(unix)]
+    #[cfg(windows)]
```

Comment:
nit: Just in case someone wants to try compiling spin to wasm some day we could make things slightly easier for them. :slightly_smiling_face: 

```suggestion
    #[cfg(not(unix))]
```